iothub/device/src/Properties.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// A container for properties.
+    /// </summary>
+    public class Properties
+    {
+        private PropertyCollection _readOnlyProperties;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="Properties"/>
+        /// </summary>
+        public Properties()
+        {
+            Writable = new PropertyCollection();
+            _readOnlyProperties = new PropertyCollection();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="Properties"/> with the specified collections
+        /// </summary>
+        /// <param name="writablePropertyCollection">A collection of writable properties returned from IoT Hub</param>
+        /// <param name="readOnlyPropertyCollection">A collection of read-only properties returned from IoT Hub</param>
+        public Properties(PropertyCollection writablePropertyCollection, PropertyCollection readOnlyPropertyCollection)
+        {
+            Writable = writablePropertyCollection;
+            _readOnlyProperties = readOnlyPropertyCollection;
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        public PropertyCollection Writable { get; private set; }
+
+        /// <summary>
+        /// Get the property from the propeties collection
+        /// </summary>
+        /// <param name="propertyName"></param>
+        /// <returns></returns>
+        public dynamic this[string propertyName]
```

Comment:
same logic is repeated within properties and propertycollection (contains, version). Should we place "readonlyprops" under a c# property as well?

Reply:
Agreed, from everything I've read and heard a property is just a property and the only real distinction comes in when it's marked as writable. So making it read only would imply that it could never be writable.